### PR TITLE
13.0 l10n ch fix postal onchange jco

### PR DESCRIPTION
**Pull request by jco-odoo**
Description of the issue/feature this PR addresses:
See if we can agree on a smaller diff for the postal onchange. 
Replacement of https://github.com/odoo/odoo/pull/51560/files

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
@yvaucher

File: addons/l10n_ch/models/res_bank.py [hunk 1]
```diff
@@ -56,27 +69,66 @@ def retrieve_acc_type(self, acc_number):
...
+        Where 09000 is the clearing number
+        """
+        return (iban.startswith('CH')
+                and iban[4:9] == CLEARING)
+
+    @api.model
+    def _pretty_postal_num(self, number):
+        """format a postal account number or an ISR subscription number
+        as per specifications with '-' separators.
+        eg. 010000628 -> 01-162-8
```
Comment (yvaucher) [inline addons/l10n_ch/models/res_bank.py:110]:
```suggestion
        eg. 010001628 -> 01-162-8
```

File: addons/l10n_ch/models/res_bank.py [hunk 2]
```diff
@@ -9,10 +9,23 @@
...
 
-from ..tools.postfinance import is_postal_num, is_postfinance_iban, iban_to_postal
+CLEARING = "09000"
 
-# backward compatibility
-_is_l10n_ch_postal = is_postal_num
+_re_postal = re.compile('^[0-9]{2}-[0-9]{1,6}-[0-9]$')
+
+def _is_l10n_ch_postal(account_ref):
+    """ Returns True iff the string account_ref is a valid postal account number,
```
Comment (yvaucher) [inline addons/l10n_ch/models/res_bank.py:17]:
```suggestion
    """ Returns True if the string account_ref is a valid postal account number,
```

**Review by yvaucher — APPROVED** [review]

**Comment by yvaucher** [comment]
@jco-odoo seems you can close this one, now that #53815 is merged

**Pull request closed without merging**